src/common/config.cpp

```diff
@@ -527,19 +537,17 @@ void load(const std::filesystem::path& path) {
 
         auto old_game_install_dir = toml::find_fs_path_or(gui, "installDir", {});
         if (!old_game_install_dir.empty()) {
-            settings_install_dirs.push_back(old_game_install_dir);
-            data.as_table().erase("installDir");
+            addGameInstallDir(old_game_install_dir);
+            gui.as_table().erase("installDir");
+            std::ofstream file(path, std::ios::out);
+            file << data;
+            file.close();
```

Comment:
This needs to be moved to the save function and done such that it's for migration ( after a major release this should be removed). Having it in the load function isn't correct.
```suggestion
            gui.as_table().erase("installDir");
            std::ofstream file(path, std::ios::out);
            file << data;
            file.close();
```

Reply:
Also this causes the rest of the setting to be not loaded if installDir exists.

Reply:
fixed in https://github.com/shadps4-emu/shadPS4/pull/1329/commits/13c1c34c82cb1b57d0316bc21c5e59a5fa1b59b4